### PR TITLE
Seed `Shuffler` for MP DataLoader without explicit `manual_seed`.

### PR DESCRIPTION
**Pull request by ejguan**
Follow up on https://github.com/pytorch/pytorch/pull/77741
See: https://github.com/pytorch/data/issues/302

This PR guarantees the `Shuffler` in first iteration with MP DataLoader has the same seed across worker processes when users don't specify the seed.
Check newly added tests

### CODE REVIEW

File: test/test_datapipe.py [hunk 1]
```diff
@@ -1501,6 +1500,14 @@ def _create_dp(buffer_size):
                     multiprocessing_context=mp_ctx,
                     worker_init_fn=_worker_init_fn
                 )
+
+                # No seed
+                dl_res_ns = list(dl)
+                self.assertEqual(len(dl_res_ns), len(exp))
+                self.assertEqual(sorted(dl_res_ns), sorted(exp))
```
Comment (ejguan) [inline test/test_datapipe.py:1507]: Even without seed, all shuffer should have the same seed/state per worker process. Then, there won't be any data loss during iteration.

**Review by NivekT — APPROVED** [review]
